keras/backend/mxnet_backend.py

```diff
@@ -2581,22 +2584,31 @@ def in_train_phase(x, alt, training=None):
         Either `x` or `alt` based on the `training` flag.
         the `training` flag defaults to `K.learning_phase()`.
     """
+    uses_learning_phase = False
+
     if training is None:
         training = learning_phase()
+        uses_learning_phase = True
 
-    if training is 1:
+    if training is 1 or training is True:
         if callable(x):
-            return x()
+            res = x()
         else:
-            return x
-    elif training is 0:
+            res = x
+        if isinstance(x, KerasSymbol):
+            uses_learning_phase = True
+    elif training is 0 or training is False:
```

Comment:
`else:`

Reply:
We can have the training parameter to be "0 or 1 or False or True or Placeholder". Hence, if, elif, else.
Simplified it more.